File: core-examples/src/main/js/io/vertx/example/core/http/simpleform/simpleformserver.js
vertx.createHttpServer().requestHandler(function (req) {
  if (req.uri() == "/") {
    // Serve the index page
    req.response().sendFile("index.html");
  } else if (req.uri().startsWith("/form")) {
    req.response().setChunked(true);
    req.setExpectMultipart(true);
    req.endHandler(function (v) {
      Array.prototype.forEach.call(req.formAttributes().names(), function(attr) {
        req.response().write("Got attr " + attr + " : " + req.formAttributes().get(attr) + "
        ");
      });
      req.response().end();
    });
  } else {
    req.response().setStatusCode(404).end();
  }
}).listen(8080);

File: core-examples/src/main/ruby/io/vertx/example/core/http/simpleform/simple_form_server.rb
$vertx.create_http_server().request_handler() { |req|
  if (req.uri().==("/"))
    # Serve the index page
    req.response().send_file("index.html")
  elsif (req.uri().start_with?("/form"))
    req.response().set_chunked(true)
    req.set_expect_multipart(true)
    req.end_handler() { |v|
      req.form_attributes().names().each do |attr|
        req.response().write("Got attr #{attr} : #{req.form_attributes().get(attr)}\n")
      end
      req.response().end()
    }
  else
    req.response().set_status_code(404).end()
  end
}.listen(8080)
